### PR TITLE
Release automation

### PR DESCRIPTION
**Pull request by anupambhatnagar**
## What does this PR do?
This PR adds a new github workflow which allows us to push releases with a click of a button. 

### Added
- `release.yml` - a new workflow called for publishing fairscale releases. **It requires a manual dispatch and the allowed input values are "major", "minor" and "patch"**
- `version.json` - this file will contain the fairscale version
- `release_utils.py` - contains some util functions used by the workflow

### Removed
- `build_wheels.yml` - a workflow we are not using 
- `manual-workflow.yml` - a mock workflow to see how manual dispatch works. no longer required

Minor edits in **setup.py**, **conf.py**, **__init__.py**

I created a mock repo to test the workflow and created a mock package that was pushed to my PyPI account. Successful runs for each of the different inputs can be seen [here](https://github.com/anupambhatnagar/package-deployer/actions/workflows/release.yml). 

## Before submitting

- [x] Did you have fun?
  - Make sure you had fun coding 🙃
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [x] N/A
- [ ] Did you make sure to update the docs?
  - [x] N/A
- [ ] Did you write any new necessary tests?
  - [x] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/main/CHANGELOG.md)? (if needed)
  - [x] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: fairscale/__init__.py [hunk 1]
```diff
@@ -3,10 +3,6 @@
 # This source code is licensed under the BSD license found in the
 # LICENSE file in the root directory of this source tree.
 
-# Please update the doc version in docs/source/conf.py as well.
-__version_tuple__ = (0, 4, 3)
-__version__ = ".".join([str(x) for x in __version_tuple__])
```
Comment (tmarkstrum) [inline fairscale/__init__.py:8]: I think __version__ here is to provide __version__ attribute for developers to use at run time. 
https://stackoverflow.com/questions/458550/standard-way-to-embed-version-into-python-package

**Review by tmarkstrum — COMMENTED** [review]
I like put version to a specific doc file. I think we should keep __version__ attribute at the same time in case developer need this attribute at run time.

File: fairscale/__init__.py (hunk 1 above)
Comment (tmarkstrum) [reply fairscale/__init__.py:8]: I propose assign a value to __version__  similar way as the set up file.

Comment (anupambhatnagar) [reply fairscale/__init__.py:8]: good catch. fixed it.

Comment (anupambhatnagar) [reply fairscale/__init__.py:8]: We need to update the value in the `__version_tuple__`. The cleanest way to do it is to move it into a separate file `version.py` containing only the `__version_tuple__` variable and update it when needed.

**Review by tmarkstrum — APPROVED** [review]
lgtm

File: fairscale/__init__.py [hunk 2]
```diff
@@ -3,16 +3,13 @@
...
-
 ################################################################################
 # Import most common subpackages
 ################################################################################
 
 from typing import List
 
-from . import nn
```
[the remark this answers is not in the document]
Comment (anupambhatnagar) [reply fairscale/__init__.py:16]: no, tests fails if this is removed.

**Pull request merged**